firebase-config/src/main/java/com/google/firebase/remoteconfig/FirebaseRemoteConfig.java

```diff
@@ -661,4 +671,51 @@ private static boolean isFetchedFresh(
       ConfigContainer fetched, @Nullable ConfigContainer activated) {
     return activated == null || !fetched.getFetchTime().equals(activated.getFetchTime());
   }
+
+  /**
+   * Starts realtime.
+   * */
+  public void startRealtime() {
+    this.configRealtimeHTTPClient.startRealtimeConnection();
+  }
+
+  /**
+   * Stops realtime.
+   * */
+  public void pauseRealtime() {
+    this.configRealtimeHTTPClient.pauseRealtimeConnection();
+  }
+
+  /**
+   * Add callback for realtime.
+   * */
+  public void addRealtimeListener(ConfigRealtimeHTTPClient.RealTimeEventListener realTimeEventListener) {
+    this.configRealtimeHTTPClient.putRealTimeEventListener("test", realTimeEventListener);
+  }
+
+  /**
+   * Starts automatic realtime handling.
+   * */
+  public void startAutomaticStreamHandling() {
+    this.automaticRealtimeHandling = true;
+  }
+
+  /**
+   * Stops automatic realtime handling.
+   * */
+  public void stopAutomaticStreamHandling() {
+    this.automaticRealtimeHandling = false;
+  }
+
+  public void handleAutomaticRealtime(boolean background) {
```

Comment:
Why do we care about background? wouldn't it be better to keep the connection open for as long as someone is listening, effectively delegating control of this to the developer?

Reply:
You probably want to re-kick the connection when the app goes in the foreground. But I agree that we should not be tearing down the connection automatically.